schema_version: 1.2.0
id: CVE-2021-29556
aliases:
- GHSA-fxqh-cfjm-fp93
modified: 2021-07-27T17:17Z
published: 2021-05-14T20:15Z
details: TensorFlow is an end-to-end open source platform for machine learning. An
  attacker can cause a denial of service via a FPE runtime error in `tf.raw_ops.Reverse`.
  This is because the implementation(https://github.com/tensorflow/tensorflow/blob/36229ea9e9451dac14a8b1f4711c435a1d84a594/tensorflow/core/kernels/reverse_op.cc#L75-L76)
  performs a division based on the first dimension of the tensor argument. The fix
  will be included in TensorFlow 2.5.0. We will also cherrypick this commit on TensorFlow
  2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as these are also
  affected and still in supported range.
severity:
- type: CVSS_V3
  score: 5.5
- type: CVSS_V2
  score: 2.1
affected:
- package:
    name: tensorflow
    cpe: cpe:2.3:a:google:tensorflow
- ranges:
  - type: GIT
    repo: https://github.com/tensorflow/tensorflow
    events:
    - introduced: '0'
    - fixed: 4071d8e2f6c45c1955a811fee757ca2adbe462c1
- database_specific:
    CWE:
    - CWE-369
    CVSS:
      Score: 5.5
      Severity: MEDIUM
      Code: CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H
references:
- type: ADVISORY
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-fxqh-cfjm-fp93
- type: FIX
  url: https://github.com/tensorflow/tensorflow/commit/4071d8e2f6c45c1955a811fee757ca2adbe462c1
